backend/api/admin/src/main/java/co/airy/core/api/admin/TagsController.java

```diff
@@ -91,12 +100,18 @@
     @PostMapping("/tags.update")
     ResponseEntity<?> updateTag(@RequestBody @Valid UpdateTagRequestPayload payload) {
         final Tag tag = stores.getTagsStore().get(payload.getId().toString());
-        if(tag == null) {
+        if (tag == null) {
             return ResponseEntity.notFound().build();
         }
 
-        tag.setColor(tagColors.get(payload.getColor()));
-        tag.setName(payload.getName());
+        if (tagColors.containsKey(payload.getColor())) {
+            tag.setColor(tagColors.get(payload.getColor()));
+        } else if (payload.getColor() != null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new RequestErrorResponsePayload(wrongTagColorErrorMessage));
+        }
+
+        tag.setName(Optional.ofNullable(payload.getName()).orElse(tag.getName()));
```

Comment:
I think of controller actions always the same way and here we'd benefit from the approach:

- verify input (if not, return error kinf of lines like in line 103)
- do the action
- return stuff

So what I'd do is test for `payload.getColor() == null || !tagColors.containsKey(payload.getColor())` and return bad request before doing the update action